Use Gson to generate the json for patches

Haerezis opened the pull request:
Hi, we talked about this in issue #4.
This is the GathererScraper part of the update. I just wanted to know if the modification I've made are ok, or if I need to refactor a little less the scraper.

If the format and the modification are ok, I'll work on mtg familiar next.
You don't need to accept the pull request (not until the mtg familiar part of the update is done and approved), I just want to know if I'm going in the right direction.

Haerezis committed:
Add a gson policy to ignore prefix in field names

The policy will ignore the prefix of attributes names and return the
rest for the field name (and force a lowercase char for the 1st of the
name)

Haerezis committed:
Update the patch format. Gson now generate them.

Gson now take care of the patches json files.
Incidently, the json format was updated to reflect the different classes
Gson is serializing to json.
So now the json format is finally human readable, and automatically
updated when the Patch,Expansion and Card classes are updated.

Haerezis committed:
Update expansions.json to match the new json format

Haerezis committed:
Remove use of RssEntry in Expansion.java

Remove the import of RssEntry and the check made in equals() method
between the expansion name and the RssEntry title.
This was done to make Expansion.java fully independant from other
classes of the project. This was donne to facilitate its export to the
mtg-familiar project, to share the core classes used in a patch, so that
gson car be used in both project, using the sames classes and code.

AEFeinstein commented:
Neat! I'm not going to merge this until after I generate the MM17 patches (should be any day now). Even then, I'll need a little time to review this.

Haerezis commented:
Ok. I'll start working on mtg-familiar even though I don't have the green light, just so I can work after on the real feature I want to do : the multi-lingual support.

Also, something to be noted, by changing the patches json format, it also changed the expansions.json file format. I've updated the file file with the new format in my branch, but it won't have MM17 in it, so you'll have to fill the infos for MM17 again after the merge (it's not a big deal, but I just wanted you to know).

AEFeinstein commented:
Thanks, I'll be aware of that.

On Mon, Mar 6, 2017 at 5:23 PM Kuntz Thomas <notifications@github.com>
wrote:

> Ok. I'll start working on mtg-familiar even though I don't have the green
> light, just so I can work after on the real feature I want to do : the
> multi-lingual support.
>
> Also, something to be noted, by changing the patches json format, it also
> changed the expansions.json file format. I've updated the file file with
> the new format in my branch, but it won't have MM17 in it, so you'll have
> to fill the infos for MM17 again after the merge (it's not a big deal, but
> I just wanted you to know).
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/AEFeinstein/GathererScraper/pull/5#issuecomment-284553692>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOHDMrYQlH46Nzb8IufLXOfYB_gzNaNks5rjId8gaJpZM4MUr5e>
> .
>

AEFeinstein committed:
Merge branch 'master' into master
